src/funjson.c

```diff
@@ -572,20 +570,24 @@ FUNCTION(fun_json)
   }
 
   if ((type == JSON_NULL && nargs > 2) ||
-      ((type == JSON_STR || type == JSON_NUMBER || type == JSON_BOOL) &&
+      ((type == JSON_STR || type == JSON_MKSTR || type == JSON_NUMBER || type == JSON_BOOL) &&
        nargs != 2) ||
       (type == JSON_OBJECT && (nargs % 2) != 1)) {
     safe_str(T("#-1 WRONG NUMBER OF ARGUMENTS"), buff, bp);
     return;
   }
 
   /* strip ansi markup from non-string types */
-  if (type != JSON_STR) {
+  if (type != JSON_STR && type != JSON_MKSTR) {
     for (i = 1; i < nargs; i++) {
       strcpy(args[i], remove_markup(args[i], NULL));
     }
   }
 
+  /* 
+    ADD A REQUIREMENT TO BE WIZ OR ABLE TO RENDER MARKUP IF JSON_MKSTR!
```

Comment:
This needs doing

Reply:
Actually not needed anymore. This documentation needs to go. It will use the standard requirements for rendering.